src/Spryker/Zed/EventBehavior/Business/Model/EventEntityTransferFilter.php

```diff
@@ -145,6 +149,101 @@ public function getEventTransfersAdditionalValues(array $eventTransfers, string
         return array_unique($additionalValues);
     }
 
+    /**
+     * @param \Generated\Shared\Transfer\HydrateEventsRequestTransfer $hydrateEventsRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\HydrateEventsResponseTransfer
+     */
+    public function hydrateEventDataTransfer(HydrateEventsRequestTransfer $hydrateEventsRequestTransfer): HydrateEventsResponseTransfer
+    {
+        $hydrateEventsResponseTransfer = new HydrateEventsResponseTransfer();
+        $sortedEventTransfers = $hydrateEventsRequestTransfer->getEventEntities()->getArrayCopy();
+
+        if (count($sortedEventTransfers) === 0) {
+            return $hydrateEventsResponseTransfer;
+        }
+        usort($sortedEventTransfers, fn (EventEntityTransfer $a, EventEntityTransfer $b) => $a->getTimestamp() <=> $b->getTimestamp());
+
+        $hydrateEventsResponseTransfer->setIdTimestampMap($this->getIdsTimestampMap($sortedEventTransfers));
+
+        if ($hydrateEventsRequestTransfer->getAdditionalValueName()) {
+            $hydrateEventsResponseTransfer->setAdditionalValueTimestampMap($this->getAdditionalValueTimestampMap($sortedEventTransfers, $hydrateEventsRequestTransfer->getAdditionalValueName()));
+        }
+
+        if ($hydrateEventsRequestTransfer->getForeignKeyName()) {
+            $hydrateEventsResponseTransfer->setForeignKeyTimestampMap($this->getForeignKeyTimestampMap($sortedEventTransfers, $hydrateEventsRequestTransfer->getForeignKeyName()));
+        }
+
+        return $hydrateEventsResponseTransfer;
```

Comment:
All of that needs to be outlined in the specification. The specification is our contract for this behavior.